docs/try-flink/table_api.md

```diff
@@ -31,482 +31,290 @@ The Table API in Flink is commonly used to ease the definition of data analytics
+    "    'driver'     = 'com.mysql.jdbc.Driver',\n" +
+    "    'username'   = 'sql-demo',\n" +
+    "    'password'   = 'demo-sql'\n" +
+    ")");
 {% endhighlight %}
-</div>
 
-<div data-lang="scala" markdown="1">
-{% highlight scala %}
-val truncateDateToHour = new TruncateDateToHour
-{% endhighlight %}
-</div>
-</div>
+The second table, `spend_report`, stores the final results of the aggregation.
+Its underlying storage is a table in a MySql database.
 
 #### The Query
 
 With the environment configured and tables registered, you are ready to build your first application.
-From the `TableEnvironment` you can `scan` an input table to read its rows and then write those results into an output table using `insertInto`.
+From the `TableEnvironment` you can read `from` an input table to read its rows and then write those results into an output table using `executeInsert`.
+The `report` function is where you will implement your business logic.
+It is currently unimplemented.
 
-<div class="codetabs" markdown="1">
-<div data-lang="java" markdown="1">
 {% highlight java %}
-tEnv
-    .scan("transactions")
-    .insertInto("spend_report");
+Table transactions = tEnv.from("transactions");
+report(transactions).executeInsert("spend_report");
 {% endhighlight %}
-</div>
 
-<div data-lang="scala" markdown="1">
-{% highlight scala %}
-tEnv
-    .scan("transactions")
-    .insertInto("spend_report")
-{% endhighlight %}
-</div>
-</div>
-
-Initially, the Job reads all transactions and logs them out with log level **INFO**.
-
-#### Execute
+## Testing 
 
-Flink applications are built lazily and shipped to the cluster for execution only once fully formed.
-You call `ExecutionEnvironment#execute` to begin the execution of your Job by giving it a name.
+The project contains a secondary testing class `SpendReportTest` that validates the logic of the report.
+It creates a table environment in batch mode. 
 
-<div class="codetabs" markdown="1">
-<div data-lang="java" markdown="1">
 {% highlight java %}
-env.execute("Spend Report");
+EnvironmentSettings settings = EnvironmentSettings.newInstance().inBatchMode().build();
+TableEnvironment tEnv = TableEnvironment.create(settings); 
 {% endhighlight %}
-</div>
 
-<div data-lang="scala" markdown="1">
-{% highlight scala %}
-env.execute("Spend Report")
-{% endhighlight %}
-</div>
-</div>
+One of Flink's unique properties is that it provides consistent semantics across batch and streaming.
+This means you can develop and test applications in batch mode on static datasets, and deploy to production as streaming applications!
 
 ## Attempt One
 
 Now with the skeleton of a Job set-up, you are ready to add some business logic.
 The goal is to build a report that shows the total spend for each account across each hour of the day.
+This means the timestamp column needs be be rounded down from millisecond to hour granularity. 
+
 Just like a SQL query, Flink can select the required fields and group by your keys.
```

Comment:
mention the term `Table API` and quickly explain what it is good for like `fluent DSL with JavaDocs and IDE support inspired by SQL`?

Reply:
yep.